sync/strategies/simple_onchange.py

```diff
@@ -46,11 +47,33 @@ def write_received_data_to_memory(self, caller, received_data):
         if received_data:
             consumed = 0
             for addr in self.addresses:
-                offset, bank_switch, mem = addr.serializer.unpack_from(received_data, consumed)
+                offset, bank_switch, should_write, mem = addr.serializer.unpack_from(received_data, consumed)
                 assert offset == addr.target_offset and bank_switch == addr.target_bank
                 consumed += addr.serializer.size
-                caller.write_memory(offset, mem, bank_switch)
-                # also update last_seen so we don't send back something that we just wrote.
                 key = (offset, bank_switch)
-                self.last_seen[key] = mem
+                if should_write:
+                    caller.write_memory(offset, mem, bank_switch)
+                    # also update last_seen so we don't send back something that we just wrote.
+                    self.last_seen[key] = mem
+                else:
+                    self.received_unsynced[key] = mem
             visualization.textlog.log_text(f'Received {self.label}')
+
+    def get_last_received_value(self, offset, length, bank_switch):
+        value = self.get_value_from_received_map(self.received_unsynced, offset, length, bank_switch)
+        if value is None:
+            value = self.get_value_from_received_map(self.last_seen, offset, length, bank_switch)
+        return value
+
+    def get_value_from_received_map(self, received_map, offset, length, bank_switch):
```

Comment:
should this be prefixed by _? (or will it be used by any foreign classes?)

Reply:
_